Upstream: no
Source: Exherbo
Reason: Need AR to be prefixed
Author: Quentin Glidic <sardemff7@exherbo.org>
diff --git a/makefile b/makefile
index f2a998e29a..14bce1225d 100644
--- a/makefile
+++ b/makefile
@@ -77,6 +77,7 @@
 # OVERRIDE_CC = cc
 # OVERRIDE_CXX = c++
 # OVERRIDE_LD = ld
+# OVERRIDE_AR = ar
 
 # DEPRECATED = 1
 # LTO = 1
@@ -499,6 +500,12 @@ ifndef CROSS_BUILD
 LD := $(OVERRIDE_LD)
 endif
 endif
+ifdef OVERRIDE_AR
+PARAMS += --AR='$(OVERRIDE_AR)'
+ifndef CROSS_BUILD
+AR := $(OVERRIDE_AR)
+endif
+endif
 
 #-------------------------------------------------
 # sanity check the configuration
diff --git a/scripts/genie.lua b/scripts/genie.lua
index e905e3b6ae..b428f0219e 100644
--- a/scripts/genie.lua
+++ b/scripts/genie.lua
@@ -188,6 +188,11 @@ newoption {
 }
 
 newoption {
+	trigger = "AR",
+	description = "AR replacement",
+}
+
+newoption {
 	trigger = "TOOLCHAIN",
 	description = "Toolchain prefix"
 }
diff --git a/scripts/toolchain.lua b/scripts/toolchain.lua
index 46a1b4b22c..81e574e2af 100644
--- a/scripts/toolchain.lua
+++ b/scripts/toolchain.lua
@@ -464,6 +464,9 @@ function toolchain(_buildDir, _subDir)
 	if (_OPTIONS["LD"] ~= nil) then
 		premake.gcc.ld  = _OPTIONS["LD"]
 	end
+	if (_OPTIONS["AR"] ~= nil) then
+		premake.gcc.ar  = _OPTIONS["AR"]
+	end
 
 	configuration {} -- reset configuration
 
